extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php

```diff
@@ -357,4 +357,22 @@ public static function onLinkEnd( $skin, Title $target, array $options, &$text,
 
 		return true;
 	}
+
+
+	/**
+	 * Hook: set script that changes the address bar in the browser for the redirected url
+	 * @param Article $page
+	 * @return bool
+	 */
+	public static function onArticleViewRedirect( Article $page ): bool {
+		if( !$page->getContext()->getUser()->isAnon() ) {
+			$script = 'history.replaceState({}, "' . $page->getTitle()->getText().'", "' .
+					  $page->getTitle()->getLocalURL() . '");';
```

Comment:
After talking about it with @Grunny:

- this is not secure, one could create a Title that closes the quotes and injects a JS
- we should not inject inline JS this here - it is better to use JS file and already existing JS variables and trigger it from php; upstream code has it implemented: https://github.com/wikimedia/mediawiki/blob/master/resources/src/mediawiki.action/mediawiki.action.view.redirect.js & https://github.com/wikimedia/mediawiki/blob/master/includes/page/Article.php#L1093-L1098; can you please try to backport it?